keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/model/ActiveServerGroup.kt

```diff
@@ -77,7 +91,24 @@ data class ServerGroup(
   override val disabled: Boolean,
   override val instanceCounts: InstanceCounts,
   override val createdTime: Long
-) : BaseEc2ServerGroup
+) : BaseEc2ServerGroup {
+  init {
+    ensureLaunchConfigInfoIsPresent(this)
+  }
+}
+
+/**
+ * Clouddriver should always return either a launchConfig field or a launchTemplate
+ * field for an EC2 server group.
+ *
+ * This is a helper function that asserts that one of these fields must be present.
+ * It is intended to be called from the init block of classes that implement BaseEc2ServerGroup
+ */
+fun ensureLaunchConfigInfoIsPresent(sg : BaseEc2ServerGroup) {
+  if(sg.launchConfig == null && sg.launchTemplate == null) {
```

Comment:
what will happen if both are not null? does clouddriver guarantees that only one of those exists?

Reply:
That's a good question. @caseyhebebrand or @jeyrschabu, does clouddriver guarantee that only one of `launchConfig` and `launchTemplate` will be returned for an EC2 server group?

Reply:
Jeyrs is the best person to answer this, but there shouldn't be case where both are present. The main reason being that a server group can only have one or the other in AWS, and we never have a state where a server group is truly modified without creating a new launch config/template.

Reply:
Both `launchConfig` and `launchTemplate` (top level props on a server group) are eventually populated as we index them. Important to note that a server group can't have both.

If a server group doesn't have a launch configuration, `launchConfigurationName` will be `null`.